app/src/examples/RuntimeTests/tests/advancedAPI/useFrameCallback.test.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import Animated, { useSharedValue, useAnimatedStyle, useFrameCallback } from 'react-native-reanimated';
+import React from 'react';
+import {
+  describe,
+  test,
+  expect,
+  render,
+  getTestComponent,
+  wait,
+  useTestRef,
+} from '../../ReanimatedRuntimeTestsRunner/RuntimeTestsApi';
+
+describe.only('Test *****useFrameCallback*****', () => {
+  describe('Test _canceling frameCallback_ after predefined time', () => {
+    const STOP_AFTER_DELAY_REF = 'CancelAfterDelayComponent';
+    const CancelAfterDelayComponent = ({ timeToStop }: { timeToStop: number }) => {
+      const width = useSharedValue(0);
+      const ref = useTestRef(STOP_AFTER_DELAY_REF);
+
+      const frameCallback = useFrameCallback(({ timeSincePreviousFrame }) => {
+        if (timeSincePreviousFrame) {
+          width.value += timeSincePreviousFrame / 8;
+        }
+      });
+
+      const animatedStyle = useAnimatedStyle(() => {
+        return { width: width.value };
+      });
+
+      useEffect(() => {
+        setTimeout(() => {
+          frameCallback.setActive(false);
+        }, timeToStop);
+      }, [timeToStop, frameCallback]);
+
+      return (
+        <View style={styles.container}>
+          <Animated.View ref={ref} style={[styles.animatedBox, animatedStyle]} />
+        </View>
+      );
+    };
+    test.each([500, 1000, 2000, 3000])('Run frameCallback for **%p**ms', async timeToStop => {
+      await render(<CancelAfterDelayComponent timeToStop={timeToStop} />);
+      const animatedComponent = getTestComponent(STOP_AFTER_DELAY_REF);
+
+      await wait(timeToStop + 200);
+
+      const expectedWidth = timeToStop / 8;
+      expect(await animatedComponent.getAnimatedStyle('width')).toBeWithinRange(expectedWidth - 3, expectedWidth + 10);
```

Comment:
What leads to such uncertainty in the result?

Reply:
setTimeout is not perfectly precise and useFrameCallback is not synchronised with setTimeout, however the range may still be reduced.

Reply:
@Latropos alright, I trust you with this one 🫡